shopinvader/services/sale.py

```diff
@@ -86,3 +86,21 @@ def _launch_notification(self, target, notif_type):
         return super(SaleService, self)._launch_notification(
             target, notif_type
         )
+
+    def _convert_one_sale(self, sale):
+        res = super(SaleService, self)._convert_one_sale(sale)
+        res["invoices"] = self._convert_invoices(sale.sudo())
+        return res
+
+    def _convert_invoices(self, sale):
+        res = []
+        for invoice in sale.invoice_ids.filtered(lambda i: i.state == "paid"):
+            res.append(self._convert_one_invoice(invoice))
+        return res
+
+    def _convert_one_invoice(self, invoice):
+        return {
+            "id": invoice.id,
+            "name": invoice.number,
+            "date": invoice.date_invoice or None,
```

Comment:
wouldn't be better to send out the URL of the invoice here?
This way, client side, we don't care to generate the URL.

Reply:
@lmignon What do you think about that?

Reply:
@simahawk Not easily since the url depends of the frontend url and we don't have this info here....

Reply:
but we do have it: on the backend you have the url of the locomotive instance, no?
Base on that you can compute the final URL. Am I missing something else?

Reply:
We have the url that will be used by odoo to call locomotive. BUT it is not necessarily the one used to navigate into the website. Moreover, on the website side, the call to odoo is done through a proxy. ("/invader"). The entry point into locomotive used to forward the calls from the fronted to odoo is not known by Odoo.

Reply:
hmm you are right. Ok, I surrender ...for now! :stuck_out_tongue_winking_eye: